fus.c

```diff
@@ -989,6 +989,57 @@ mask_non_default_module_pkgs (Pool *pool)
   return selection;
 }
 
+/*
+ * Mask bare rpms if any of the default modules provides them (even if older)
+ */
+static Queue
+mask_bare_rpms (Pool *pool, Map *modular_pkgs)
+{
+  Queue selection;
+  queue_init (&selection);
+
+  Id def_modules_rel = pool_rel2id (pool,
+                                    pool_str2id (pool, "module()", 1),
+                                    pool_str2id (pool, "module-default()", 1),
+                                    REL_WITH,
+                                    1);
+
+  Id *pp = pool_whatprovides_ptr (pool, def_modules_rel);
+  for (; *pp; pp++)
+    {
+      Solvable *s = pool_id2solvable (pool, *pp);
+      g_auto(Queue) q;
+      queue_init (&q);
+      Id dep = pool_rel2id (pool, s->name, s->arch, REL_ARCH, 1);
+      pool_whatcontainsdep (pool, SOLVABLE_REQUIRES, dep, &q, 0);
+
+      for (int i = 0; i < q.count; i++)
+        {
+          /* Extract name from rpm */
+          const char *nevra = pool_solvid2str (pool, q.elements[i]);
+          unsigned int name_len = strchr (nevra, '-') - nevra;
```

Comment:
Is this going to work with package names that contain `-`? Maybe it would be safer to get a `Solvable` for the package and get `s->name` from there.

Reply:
No, it would not. I went with the dirtiest approach just to get some feedback. I'll change it to the approach you suggested.